wasatch/OceanDevice.py

```diff
@@ -0,0 +1,181 @@
+            # about. That would actually provide a reason to roll all the
+            # temperature etc readouts into the SpectrometerState class...
+            reading.integration_time_ms = self.settings.state.integration_time_ms
+            reading.laser_power_perc    = self.settings.state.laser_power_perc
+            reading.laser_power_mW      = self.settings.state.laser_power_mW
+            reading.laser_enabled       = self.settings.state.laser_enabled
+            reading.spectrum = list(self.spec.intensities())
+
+            if not reading.failure:
+                if averaging_enabled:
+                    if self.sum_count == 0:
+                        self.summed_spectra = [float(i) for i in reading.spectrum]
+                    else:
+                        log.debug("device.take_one_averaged_reading: summing spectra")
+                        for i in range(len(self.summed_spectra)):
+                            self.summed_spectra[i] += reading.spectrum[i]
+                    self.sum_count += 1
+                    log.debug("device.take_one_averaged_reading: summed_spectra : %s ...", self.summed_spectra[0:9])
+
+            # count spectra
+            self.session_reading_count += 1
+            reading.session_count = self.session_reading_count
+            reading.sum_count = self.sum_count
+
+            # have we completed the averaged reading?
+            if averaging_enabled:
+                if self.sum_count >= self.settings.state.scans_to_average:
+                    reading.spectrum = [ x / self.sum_count for x in self.summed_spectra ]
+                    log.debug("device.take_one_averaged_reading: averaged_spectrum : %s ...", reading.spectrum[0:9])
+                    reading.averaged = True
+
+                    # reset for next average
+                    self.summed_spectra = None
+                    self.sum_count = 0
+            else:
+                # if averaging isn't enabled...then a single reading is the
+                # "averaged" final measurement (check reading.sum_count to confirm)
+                reading.averaged = True
+
+            # were we told to only take one (potentially averaged) measurement?
+            if self.take_one and reading.averaged:
+                log.debug("completed take_one")
+                self.change_setting("cancel_take_one", True)
+
+        log.debug("device.take_one_averaged_reading: returning %s", reading)
+        # reading.dump_area_scan()
+        return reading
+
+    def change_setting(self,setting,value):
+        if setting == "scans_to_average":
+            self.sum_count = 0
+            self.settings.state.scans_to_average = int(value)
+            return
+        f = self.lambdas.get(setting, None)
+        if f is None:
+            # quietly fail no-ops
+            return False
+
+        log.info(f"about to set integation time using func {f} to value {value}")
```

Comment:
Okay, I don't think I like this.  This is ASSUMING that the only lambda we support is set_integration_time_ms.  Otherwise this would get called for EVERY possible settings, including those where the parameter should not be scaled up by an order of magnitude.